src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/viewmodels/ReportDraftViewModel.kt

```diff
@@ -0,0 +1,57 @@
+package org.vaccineimpact.orderlyweb.viewmodels
+
+import org.vaccineimpact.orderlyweb.db.AppConfig
+import org.vaccineimpact.orderlyweb.models.ReportDraft
+import org.vaccineimpact.orderlyweb.models.ReportWithPublishStatus
+import java.time.Instant
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+
+data class ReportDraftViewModel(val id: String,
+                                val url: String,
+                                val changelog: List<ChangelogItemViewModel>,
+                                val parameterValues: String?)
+{
+    companion object
+    {
+        fun build(version: ReportDraft): ReportDraftViewModel
+        {
+            val changelogs = version.changelogs.map { ChangelogItemViewModel.build(it) }
+            val parameterValues = version.parameterValues.entries.joinToString(",") { "${it.key}=${it.value}" }
+            return ReportDraftViewModel(version.id, "${AppConfig()["app.url"]}/${version.name}/${version.id}", changelogs, parameterValues)
+        }
+    }
+}
+
+
+data class DateGroup(val date: String, val drafts: List<ReportDraftViewModel>)
```

Comment:
Is a date group not the same as a batch then? I guess since batches are new, no current report has a batch. But would idea be to transfer these drafts over to being organised by batch (if any) rather than by date? - quite likely could have multiple batches of same report run on a single day (display string could be first run time in batch).

Reply:
So in the mockup I didn't explicitly group by batch but rather by date. I think it's useful to group by date to make the page more readable and probably enough as another level of nesting (date -> batch) might be less readable and in practice batched reports will have run times v close to each other, so appear in order. But I will go back to Kim and confirm these points.